sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -298,6 +298,13 @@ class SQLQuerySuite extends QueryTest with BeforeAndAfterAll {
     )
   }
 
+  test("left semi greater than predicate and equal operator") {
```

Comment:
@adrian-wang i suggest you add the case chenghao described in my PR to the unit test.